src/strategies/build_strategy.rs

```diff
@@ -0,0 +1,413 @@
+use std::{env, path::PathBuf, process::Command};
+
+use anyhow::{anyhow, bail, Result};
+use blue_build_utils::constants::*;
+use log::{info, trace};
+use uuid::Uuid;
+
+#[cfg(feature = "builtin-podman")]
+use anyhow::Context;
+#[cfg(feature = "futures-util")]
+use futures_util::StreamExt;
+#[cfg(feature = "builtin-podman")]
+use log::{debug, error};
+#[cfg(feature = "podman-api")]
+use podman_api::{
+    opts::{
+        ContainerListOpts, ContainerPruneFilter, ContainerPruneOpts, ImageBuildOpts,
+        ImagePruneFilter, ImagePruneOpts, ImagePushOpts, ImageTagOpts, RegistryAuth,
+    },
+    Podman,
+};
+#[cfg(feature = "signal-hook")]
+use signal_hook::consts::{SIGHUP, SIGINT, SIGQUIT, SIGTERM};
+#[cfg(feature = "signal-hook-tokio")]
+use signal_hook_tokio::Signals;
+#[cfg(feature = "builtin-podman")]
+use std::sync::Arc;
+#[cfg(feature = "tokio")]
+use tokio::{
+    runtime::Runtime,
+    sync::oneshot::{self, Sender},
+    time::{self, Duration},
+};
+
+use crate::commands::build::Credentials;
+
+#[derive(Debug)]
+enum Strategy {
+    Buildah,
+    Podman,
+    #[cfg(feature = "builtin-podman")]
+    Socket(Arc<Podman>, Runtime),
```

Comment:
This might need to include docker as well. A generic inside arc might be needed

Reply:
No that won't work in this case as this is a specific Podman Api variant with code pertaining to its library.